source/features/raw-file-link.tsx

```diff
@@ -4,28 +4,26 @@ import delegate from 'delegate-it';
 import * as pageDetect from 'github-url-detection';
 
 import features from '.';
+import GitHubURL from '../github-helpers/github-url';
 
 function handleMenuOpening(event: delegate.Event): void {
+	event.delegateTarget.classList.add('rgh-raw-file-link'); // Mark this as processed
 	const dropdown = event.delegateTarget.nextElementSibling!;
 
-	// Only if it's not already there
-	if (select.exists('.rgh-raw-file-link', dropdown)) {
-		return;
-	}
-
 	const viewFile = select<HTMLAnchorElement>('[data-ga-click^="View file"]', dropdown)!;
-	const url = viewFile.pathname.split('/');
-	url[3] = 'raw'; // Replaces 'blob'
+	const href = new GitHubURL(viewFile.href).assign(
+		{route: 'raw'}
+	).toString();
```

Comment:
Can be one line. Also you can drop `toString` and just use destructuring:

const {href} = etc